pkg/controller/controller.go

```diff
@@ -162,6 +167,19 @@ func (r *LightHouseController) removeServiceWatcher(clusterID string) {
 	}
 }
 
+func (r *LightHouseController) handleUpdateService(old interface{}, newObj interface{}, remoteCluster *remoteCluster) {
+	var key string
+	var err error
+	if key, err = cache.MetaNamespaceKeyFunc(newObj); err != nil {
+		return
+	}
+	oldGlobalIp := old.(*corev1.Service).GetAnnotations()[submarinerIpamGlobalIp]
+	newGlobalIp := newObj.(*corev1.Service).GetAnnotations()[submarinerIpamGlobalIp]
```

Comment:
In both cases you should retrieve the value by key with care. Doing like this (instead of the form `value, ok =` is risky, no? 

Unless I'm missing something, should the key not exist this will panic and kill the execution.

Reply:
I thhought that was an issue only when handling deletes. Is it possible on create/update too?

Reply:
what I mean is that `GetAnnotations()` returns a `map` and then you are getting the value to store it in `xxxGlobalIp` using `submarinerIpamGlobalIp` as a key. If `GetAnnotations` returns a `map` that does not contain that key, then the get panics. That's why if there is a chance that it won't have the key you should use the getter that also returns an `ok` value.

Reply:
It doesn't panic. All services will never have the annotation when service is first created. If it caused a panic this code won't have worked at all.